src/aspire/basis/fpswf_2d.py

```diff
@@ -1,5 +1,300 @@
+        functions on a disk", Appl. Comput. Harmon. Anal. 43, 381-403 (2017).
+        3) Yoel Shkolnisky, "Prolate spheroidal wave functions on a disc-Integration and approximation of
+        two-dimensional bandlimited functions", Appl. Comput. Harmon. Anal. 22, 235-256 (2007).
+    """
+
+    def __init__(self, size, gamma_truncation=1.0, beta=1.0):
+        """
+        Initial an object for 2D Prolate Spheroidal Wave Function (PSWF) basis expansion using fast method.
+        """
+        super().__init__(size, gamma_truncation, beta)
+
+    def _build(self):
+        """
+        Build internal data structures for the direct 2D PSWF method.
+        """
+        logger.info('Expanding 2D images using fast PSWF method.')
+
+        # initial the whole set of PSWF basis functions based on the bandlimit and eps error.
+        self.bandlimit = self.beta * np.pi * self.rcut
+        self.d_vec_all, self.alpha_all, self.lengths = self.init_pswf_func2d(self.bandlimit, eps=np.spacing(1))
+
+        # generate_the 2D grid and corresponding indices inside the disc.
+        self.generate_grid()
+
+        # precompute the basis functions in 2D grids
+        self.precomp()
+
+    def precomp(self):
+        """
+        Precomute the basis functions on a polar Fourier 2D grid.
+        """
+        self.generate_samples()
+
+        eps = np.spacing(1)
+        a, b, c, d, e, f = self._generate_pswf_quad(4 * self.rcut, 2 * self.bandlimit, eps, eps, eps)
+        self.pswf_radial_quad = self.evaluate_pswf2d_all(d, np.zeros(len(d)), self.max_ns)
+        self.quad_rule_pts_x = a
+        self.quad_rule_pts_y = b
+        self.quad_rule_wts = c
+        self.radial_quad_pts = d
+        self.quad_rule_radial_wts = e
+        self.num_angular_pts = f
+
+        # pre computing variables for forward
+        us_fft_pts = np.column_stack((self.quad_rule_pts_x, self.quad_rule_pts_y))
+        us_fft_pts = self.bandlimit / (self.rcut * np.pi * 2) * us_fft_pts  # for pynfft
+        blk_r, num_angular_pts, r_quad_indices, numel_for_n, indices_for_n, n_max =\
+            self._pswf_integration_sub_routine()
+
+        self.us_fft_pts = us_fft_pts
+        self.blk_r = blk_r
+        self.num_angular_pts = num_angular_pts
+        self.r_quad_indices = r_quad_indices
+        self.numel_for_n = numel_for_n
+        self.indices_for_n = indices_for_n
+        self.n_max = n_max
+        self.size_x = len(self.points_in_disk)
+
+    def evaluate_t(self, images):
+        """
+        Evaluate coefficient vectors in PSWF basis using the fast method
+
+        :param images: coefficient array in the standard 2D coordinate basis to be evaluated.
+        :return : The evaluation of the coefficient array in the PSWF basis.
+        """
+        # start and finish are for the threads option in the future
+        images_shape = images.shape
+        start = 0
+        # if we got several images
```

Comment:
Move this comment inside `if` clause (same for comment preceding the `else` clause).

Reply:
Done.